### DISCUSSION TITLE
Keycloak cannot create user in LDAP

### DISCUSSION THREAD

**Discussion by werwuifi** [keycloak/keycloak#23565] [Q&A]
I'm encountering an issue while using the Keycloak Admin API (Keycloak version 22.0.3). I've been able to successfully retrieve users, roles, groups, and other data. However, when I attempt to create or update a user, I'm receiving an error, specifically, '{"error": "unknown_error"},' or I see the following error in the container logs: 'Caused by: javax.naming.AuthenticationException: [LDAP: error code 49 - 80090308: LdapErr: DSID-0C090447, comment: AcceptSecurityContext error, data 52e, v3839].'

I've also posted this question [here](https://stackoverflow.com/questions/77182024/keycloak-cannot-create-user-in-ldap). Can anyone provide insights or suggestions on how to resolve this issue?

**Comment by justin-tay** [comment]
Are the service account credentials specified in `Bind DN` and `Bind credentials` correct? Does the `Test authentication` button work?

**Reply by werwuifi** [reply]
The "test authentication" button works ("Successfully connected to LDAP"), however I figured out that searching users fails due to a misconfigured identity provider ("No users found, could be due to wrongly configured federated provider").

I removed the erroneous provider config and the LDAP error I got when creating a new user changed to '{ "errorMessage": "Could not create user" }' or "javax.naming.directory.InvalidAttributeValueException: [LDAP: error code 21 - 00000057: LdapErr: DSID-0C0910E6, comment: Error in attribute conversion operation, data 0, v3839]; remaining name 'cn=username,OU=User,OU=company,DC=domain,DC=at'"

So it's still not working, there's still an error, just the root cause is different. I checked the LDAP schema for the username (Directory String). Is this still keycloak related? I have little knowledge of LDAP configuration issues.

**Reply by werwuifi** [reply]
Any idea what could be the issue?
